src/Microsoft.Health.Encryption/Customer/Health/CustomerKeyValidationBackgroundService.cs

```diff
@@ -0,0 +1,77 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using EnsureThat;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Core.Features.Health;
+
+namespace Microsoft.Health.Encryption.Customer.Health;
+
+internal class CustomerKeyValidationBackgroundService : BackgroundService
+{
+    private const string AccessLostMessage = "Access to the customer-managed key has been lost";
+
+    private readonly IKeyTestProvider _keyTestProvider;
+    private readonly AsyncData<CustomerKeyHealth> _customerManagedKeyHealth;
+    private readonly ILogger<CustomerKeyValidationBackgroundService> _logger;
+
+    public CustomerKeyValidationBackgroundService(
+        IKeyTestProvider keyTestProvider,
+        AsyncData<CustomerKeyHealth> customerManagedKeyHealth,
+        ILogger<CustomerKeyValidationBackgroundService> logger)
+    {
+        _keyTestProvider = EnsureArg.IsNotNull(keyTestProvider, nameof(keyTestProvider));
+        _customerManagedKeyHealth = EnsureArg.IsNotNull(customerManagedKeyHealth, nameof(customerManagedKeyHealth));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                await CheckHealth(stoppingToken).ConfigureAwait(false);
+                await Task.Delay(TimeSpan.FromSeconds(60), stoppingToken).ConfigureAwait(false);
```

Comment:
Should this be configurable in `CustomerManagedKeyOptions`?

Reply:
Added new property `KeyValidationPeriodInSeconds`, but left 60 seconds as the default